### PR TITLE
Converters clean up

### PR DESCRIPTION
**Pull request by dunqan**
1. Opt-out from using list normalizers in favor of singular ones.
2. Use UICart model and proper normalizer
3. Don't use product image normalizer directly

Closes GH-2034

### CODE REVIEW

File: projects/core/src/product/occ/converters/product-image-normalizer.spec.ts [hunk 1]
```diff
@@ -108,12 +109,8 @@ describe('ProductImageConverterService', () => {
...
   it('should convert product image', () => {
-    service.convertProduct(product);
-    expect(product).toBeTruthy(convertedProduct);
-  });
-
-  it('should convert a list of products', () => {
-    service.convertList([product]);
-    expect(product).toBeTruthy(convertedProduct);
+    const result = service.convert(product);
+    console.log(result);
```
Comment (Platonn) [inline projects/core/src/product/occ/converters/product-image-normalizer.spec.ts]: please remove console.log

File: projects/core/src/cart/store/effects/cart.effect.ts [hunk 2]
```diff
@@ -72,7 +69,10 @@ export class CartEffects {
           switchMap((cart: Cart) => {
             if (cart.entries) {
               for (const entry of cart.entries) {
-                this.productImageConverter.convertProduct(entry.product);
+                entry.product = this.converter.convert(
```
Comment (Platonn) [inline projects/core/src/cart/store/effects/cart.effect.ts]: I'm not sure why we use `PRODUCT_NORMALIZER` in the cart effect. Maybe we can use CartNormalizer?

Comment (dunqan) [reply projects/core/src/cart/store/effects/cart.effect.ts]: Very good point, thanks!

File: projects/core/src/product/occ/converters/product-image-normalizer.spec.ts (hunk 1 above)
Comment (dunqan) [reply projects/core/src/product/occ/converters/product-image-normalizer.spec.ts]: 👍

**Pull request merged**